test/unit/snappy.test.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const mock = require('../tools/mock');
+const snappy = optionalRequire('snappy');
+const snappyVersion = optionalRequire('snappy/package.json').version;
+
+function optionalRequire(mod) {
+  try {
+    return require(mod);
+  } catch {
+    return false;
+  }
+}
+
+describe('Compression', function () {
+  let client;
+  /** @type {mock.MockServer} */
+  let server;
+  before(async function () {
+    if (!snappy) this.skip();
+    server = await mock.createServer();
+    client = this.configuration.newClient(`mongodb://${server.uri()}`, { compressors: 'snappy' });
+  });
+
+  after(async function () {
+    await mock.cleanup();
+    await client.close();
+  });
+
+  describe('Snappy', () => {
+    it(`should compress messages sent with snappy ${snappyVersion}`, async function () {
+      // This test should not take any longer
+      // The failure case is hanging because new snappy never calls the callback
```

Comment:
what failure case?

Reply:
I think maybe these two comments mean that the timeout is being set because the test shouldn't take any longer than 5 seconds, and that if it doesn't complete, it will hang due to the callback never being called, is that right?

Reply:
Yep that's all correct, I can clarify all that.